hoomd/md/methods/methods.py

```diff
@@ -30,7 +32,104 @@ def _detach_hook(self):
         self._simulation.state.update_group_dof()
 
 
-class NVT(Method):
+class Thermostat(_HOOMDBaseObject):
```

Comment:
I see you went with the `_HOOMDBaseObject` approach. This more closely matches the `NeighborList` model. Given the complexity of these classes, this seems right. @b-butler do you agree?

Reply:
Since we need to deal with a C++ object that requires outside information, this is the right approach.